### PR TITLE
Feat/refactor/environmental variables

### PR DESCRIPTION
**Pull request by Sgalvin36**
## Type of Change
- [x] feature ⛲
- [x] bug fix 🐛

## Description

Added environmental variables specifically for the api calls so that it utilizes the correct URL depending on if it is the test, development, or production branch of the application. No new dependencies, it relies on how you start the application. `npm start` will utilize the development variables, while `npm build` will utilize the production variables. Cypress will default utilize the development variables as well unless explicitly told to use the test ones.

Refactored all backend API calls into single file that was renamed `trackerApiCalls`. Though the use of the environmental variables helps with API calls, it is still helpful to have them all in one place. Major change to Login component and minor bug fix to userInformation component that is tied with the api calls and the specifics of the changed JSON contract.
## Motivation and Context

[Issue 21](https://github.com/turingschool/tracker-crm-fe/issues/21)
It solves an important issue that happens as an application moves from development to deployment. Instead of needing to locate and change all `localhost` url's to the deployed url, using environmental variables allows us to define those based off the specific branch of the application that is being used. The development branch will utilize `localhost` while the deployment branch will specifically target the deployed backend API.
## Related Tickets

[Issue 55](https://github.com/turingschool/tracker-crm/issues/55)
While validating that the refactored API call continued to work, changes to the way the front end was reading the JSON contract needed to be made so that all tests worked as intended. This should address the bug that was mentioned in issue 55 as well.

## Added Test?
- [ ] Yes 🫡
  - 
- [ ] No 🙅
- [x] All previous tests still pass 🥳

## Checklist:

- [x] My code follows the code style of this project.

### CODE REVIEW

File: src/trackerApiCalls.tsx
Comment (jimmacur) [inline src/trackerApiCalls.tsx:1]: This is a great addition. I am glad to see us using this along with the environmental variables!

**Review by jimmacur — COMMENTED** [review]
This is looking great. Currently this change makes 3 areas of tests fail - landing spec, companies and newCompany. Not sure if that should be handled on this ticket or make a new one.

**Review by ROlearyPro — APPROVED** [review]
This is quite useful in terms of how we're going to be handling the URLs, both in testing and otherwise. Similarly, I have to agree with Jim, the API calls being condensed into a single file will help us keep track of everything much more easily. 
I noticed that there's errors for each cypress test file at the moment, but I don't think that's so much your fault as the tests still using localhost and such. We'll have to fix that, but that can be a separate issue. Nice work!

**Comment by jimmacur** [comment]
I agree with Ryan that tests can become their own ticket.

**Review by jimmacur — APPROVED** [review]

**Pull request merged**